compiler/rustc_smir/src/stable_mir/mod.rs

```diff
@@ -13,11 +13,15 @@
 
 use std::cell::Cell;
 
-use crate::rustc_smir::Tables;
-
 use self::ty::{
     GenericDef, Generics, ImplDef, ImplTrait, PredicateKind, Span, TraitDecl, TraitDef, Ty, TyKind,
 };
+use crate::rustc_internal;
+use crate::rustc_smir::Tables;
+use rustc_driver::{Callbacks, Compilation, RunCompiler};
+use rustc_interface::{interface, Queries};
+use rustc_middle::ty::TyCtxt;
+use rustc_session::EarlyErrorHandler;
```

Comment:
We should not have any `rustc_*` stuff inside the `stable_mir` modules. Everything should be abstracted via the `Context` trait. Otherwise we will not be able to build the SMIR crate on stable on crates.io in the future.

In this case, we cannot abstract it, because in order to get a `Context` instance, we first need to start up rustc. Let's figure out how to handle this best separately. for now, move this logic to where `rustc_internal::run` lives.